common/big_messaging/src/main/scala/uk/ac/wellcome/bigmessaging/VHS.scala

```diff
@@ -42,3 +49,32 @@ class VHSInternalStore[T, Metadata](
       id.version.toString,
       UUID.randomUUID().toString)
 }
+
+/** There are times when we want to be able to abstract away the fact that we
+  * are using a VHS[T, EmptyMetadata], and instead use the more general Store
+  * interface. However the standard VHS is a leaky abstraction, as it exposes
+  * the fact that internally it uses a hybrid storage method, returning a
+  * HybridStoreEntry[T, EmptyMetadata] rather than plainly T.
+  *
+  * Here we wrap the VHS to provide an interface that returns some item T
+  * rather than a HybridStoreEntry[T, EmptyMetadata]
+  */
+class VHSWrapper[T](vhs: VHS[T, EmptyMetadata])
+    extends Store[Version[String, Int], T]
+    with Maxima[String, Int] {
+
+  def get(id: Version[String, Int]): ReadEither =
+    vhs.get(id).map {
+      case Identified(key, HybridStoreEntry(item, _)) =>
+        Identified(key, item)
+    }
+
+  def put(id: Version[String, Int])(item: T): WriteEither =
+    vhs.put(id)(HybridStoreEntry(item, EmptyMetadata())).map {
+      case Identified(key, HybridStoreEntry(item, _)) =>
+        Identified(key, item)
+    }
+
+  def max(q: String): Either[ReadError, Int] =
+    vhs.hybridStore.max(q)
+}
```

Comment:
This is nice, doesn't have tests though (and can't think of what the tests might be)?